include/fbgemm/PackingTraits-inl.h

```diff
@@ -6,6 +6,41 @@
  */
 #pragma once
 
+/*
+ * This file configures the important cache blocking parameters and registers
+ * blocking parameters for the matrix multiplication loops inside FBGEMM.
+ *
+ * ROW_INTERLEAVE: the number of interleaved rows to use vpmaddubsw instructions
+ * for packing B matrix. For 32-bit accumulation, ROW_INTERLEAVE = 4; For 16-bit
+ * accumulation, ROW_INTERLEAVE = 2.
+ *
+ * VLEN: the vector length of one SIMD register. For avx2, VLEN = 256; For
+ * avx512, VLEN = 512.
+ *
+ * NR: the register blocking parameters for N dimension. NR columns of
+ * interleaved rows of int8 (singed or unsigned) should fit into one SIMD
+ * register. Basically, NR = VLEN/ 8 / ROW_INTERLEAVE (8 is the bit length for
+ * int8 (signed or unsigned).
+ *
+ * MR: the register blocking parameters for M dimension. MR is the total number
+ * of SIMD registers used for M dimension of registers used for accumulation C.
+ * This indicates the number of vpbroadcastw instructions for A.
+ *
+ * NCB: the cache blocking parameters for N dimension. NCB needs to be multiple
+ * of NR. The total register on N dimension of registers used for
+ * accumulation C should be NCB/NR.
+ *
+ * (MR) * (NC/NR): the number of registers used for accumulation C. (MR) *
```

Comment:
Ideally this should be MR * NR = the number of registers used for C result accumulation. We don't have an inner loop in the kernel of size NCB/NR. That would free NCB totally is cache blocking parameter. Right now ... it can't be too large. For now it's fine until we update the inner kernel.

Reply:
Right. I realized that we don't have a real cache blocking parameter on the N dimension.